self_play.py

```diff
@@ -245,6 +247,13 @@ def select_action(node, temperature):
 
         return action
 
+def select_random(list_):
+    """ 
+    Random selection of an element from a list.
+    Faster than numpy.random.choice
+    
+    """
+    return list_[int(len(list_) * random.random())] 
```

Comment:
What hapens here is a substitute of random.randint( 0, len(list_) ), since randint is quite slow. 
An even faster operation would be to use math.floor(...) instead of int(...), although they do the same functionality.